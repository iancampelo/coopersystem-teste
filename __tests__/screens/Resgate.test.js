import 'react-native'
import React from 'react'
import ResgateScreen from '../../src/screens/Resgate'
import renderer from 'react-test-renderer'

jest.useFakeTimers()
jest.mock('modal-enhanced-react-native-web', () => props => <div {...{ ...props }} />)

const mockAcoes = {
  nome: 'INVESTIMENTO I',
  objetivo: 'Minha aposentadoria',
  saldoTotalDisponivel: 39321.29,
  indicadorCarencia: 'N',
  acoes: [
    {
      id: '1',
      nome: 'BBAS3',
      percentual: 28.1,
    },
    {
      id: '2',
      nome: 'VALE3',
      percentual: 20.71,
    },
    {
      id: '3',
      nome: 'PETR4',
      percentual: 21.63,
    },
    {
      id: '4',
      nome: 'CMIG3',
      percentual: 12.41,
    },
    {
      id: '5',
      nome: 'OIBR3',
      percentual: 17.15,
    },
  ],
};

it('renders correctly ResgateScreen', () => {
  const push = jest.fn();
  const tree = renderer.create(
    <ResgateScreen
      navigation={{ push }}
      route={{
        valorInvestimentos: mockAcoes.saldoTotalDisponivel,
        acoes: mockAcoes.acoes,
        nome: mockAcoes.nome
      }}
    />).toJSON()
  expect(tree).toMatchSnapshot()
})
