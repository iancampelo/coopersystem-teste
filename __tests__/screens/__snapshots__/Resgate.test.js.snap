// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly ResgateScreen 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <div
    isVisible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 22,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 0,
            "elevation": 5,
            "margin": 20,
            "padding": 35,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.5,
            "shadowRadius": 3.84,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 10,
            }
          }
        >
          RESGATE EFETUADO!
        </Text>
        <Text
          style={
            Object {
              "marginBottom": 15,
              "textAlign": "center",
            }
          }
        >
          O valor solicitado estará em sua conta em até 5 dias úteis!
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#fae128",
              "elevation": 2,
              "marginTop": 10,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#005aa5",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            NOVO RESGATE
          </Text>
        </View>
      </View>
    </View>
  </div>
  <Text
    style={
      Object {
        "color": "gray",
        "fontWeight": "bold",
        "padding": 15,
      }
    }
  >
    DADOS DO INVESTIMENTO
  </Text>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Nome
      </Text>
      <Text />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Saldo total disponível
      </Text>
      <Text>
        
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "gray",
        "fontWeight": "bold",
        "padding": 15,
      }
    }
  >
    RESGATE DO SEU JEITO
  </Text>
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <Text
    style={
      Object {
        "color": "gray",
        "fontWeight": "bold",
        "padding": 15,
      }
    }
  >
    Valor total a resgatar 
    
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "gray",
        "elevation": 2,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#005aa5",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      CONFIRMAR RESGATE
    </Text>
  </View>
</View>
`;
